src/query/exists_query.rs

```diff
@@ -0,0 +1,183 @@
+use common::BitSet;
+use crate::query::{BitSetDocSet, ConstScorer, EnableScoring, Explanation, Query, Scorer, Weight};
+use crate::query::explanation::does_not_match;
+use crate::{DocId, Score, SegmentReader, Result};
+use crate::schema::{Field, IndexRecordOption};
+
+/// An Exists Query matches all of the documents
+/// containing a specific indexed field.
+///
+/// ```rust
+/// use tantivy::collector::Count;
+/// use tantivy::query::ExistsQuery;
+/// use tantivy::schema::{Schema, TEXT};
+/// use tantivy::{doc, Index};
+///
+/// # fn test() -> tantivy::Result<()> {
+/// let mut schema_builder = Schema::builder();
+/// let title = schema_builder.add_text_field("title", TEXT);
+/// let author = schema_builder.add_text_field("author", TEXT);
+/// let schema = schema_builder.build();
+/// let index = Index::create_in_ram(schema);
+/// {
+///     let mut index_writer = index.writer(3_000_000)?;
+///     index_writer.add_document(doc!(
+///         title => "The Name of the Wind",
+///         author => "Patrick Rothfuss"
+///     ))?;
+///     index_writer.add_document(doc!(
+///         title => "The Diary of Muadib",
+///     ))?;
+///     index_writer.add_document(doc!(
+///         title => "A Dairy Cow",
+///         author => "John Webster"
+///     ))?;
+///     index_writer.commit()?;
+/// }
+///
+/// let reader = index.reader()?;
+/// let searcher = reader.searcher();
+///
+/// let query = ExistsQuery::new(author);
+/// let count = searcher.search(&query, &Count)?;
+/// assert_eq!(count, 2);
+/// Ok(())
+/// # }
+/// # assert!(test().is_ok());
+/// ```
+#[derive(Clone, Debug)]
+pub struct ExistsQuery {
+    field: Field,
+}
+
+impl ExistsQuery {
+    /// Creates a new ExistsQuery with a given field
+    pub fn new(field: Field) -> Self {
+        ExistsQuery { field }
+    }
+}
+
+impl Query for ExistsQuery {
+    fn weight(&self, _: EnableScoring<'_>) -> Result<Box<dyn Weight>> {
+        Ok(Box::new(ExistsWeight {
+            field: self.field
+        }))
+    }
+}
+
+/// Weight associated with the `ExistsQuery` query.
+pub struct ExistsWeight {
+    field: Field,
+}
+
+impl Weight for ExistsWeight {
+    fn scorer(&self, reader: &SegmentReader, boost: Score) -> Result<Box<dyn Scorer>> {
+        let max_doc = reader.max_doc();
+        let mut doc_bitset = BitSet::with_max_value(max_doc);
+
+        let inverted_index = reader.inverted_index(self.field)?;
+        let term_dict = inverted_index.terms();
+        let mut term_stream = term_dict.range().into_stream()?;
```

Comment:
Isn't this equivalent to

```rust
let mut term_stream = term_dict.stream()?;
```

Reply:
yes, I didn't see that function, have simplified